src/dex/generic-rfq/rate-fetcher.ts

```diff
@@ -0,0 +1,387 @@
+import BigNumber from 'bignumber.js';
+import { SwapSide } from 'paraswap-core';
+import { BN_0, BN_1, getBigNumberPow } from '../../bignumber-constants';
+import { IDexHelper } from '../../dex-helper';
+import Fetcher from '../../lib/fetcher/fetcher';
+import { Logger, Address, Token } from '../../types';
+import { OrderInfo } from '../paraswap-limit-orders/types';
+import {
+  BigNumberRate,
+  BigNumberRates,
+  MarketResponse,
+  OrderPriceInfo,
+  PairMap,
+  PriceAndAmount,
+  PriceAndAmountBigNumber,
+  Rates,
+  RatesResponse,
+  RFQConfig,
+  RFQFirmRateResponse,
+  RFQPayload,
+} from './types';
+
+export const FIRM_RATE_TIMEOUT_MS = 500;
+
+function onlyUnique(value: string, index: number, self: string[]) {
+  return self.indexOf(value) === index;
+}
+
+export const reversePrice = (price: PriceAndAmountBigNumber) =>
+  ({
+    amount: price.amount.times(price.price),
+    price: BN_1.dividedBy(price.price),
+  } as PriceAndAmountBigNumber);
+
+export class RateFetcher {
+  private augustusAddress: Address;
+
+  private marketFetcher: Fetcher<MarketResponse>;
+  private rateFetcher: Fetcher<RatesResponse>;
+
+  private pairs: PairMap = {};
+
+  constructor(
+    private dexHelper: IDexHelper,
+    private config: RFQConfig,
+    private dexKey: string,
+    private logger: Logger,
+  ) {
+    (this.augustusAddress =
+      dexHelper.config.data.augustusAddress.toLowerCase()),
+      (this.marketFetcher = new Fetcher<MarketResponse>(
+        dexHelper.httpRequest,
+        {
+          info: {
+            requestOptions: config.marketConfig.reqParams,
+            caster: this.castMarketResponse.bind(this),
+          },
+          handler: this.handleMarketResponse.bind(this),
+        },
+        config.marketConfig.intervalMs,
+        this.logger,
+      ));
```

Comment:
Why these expressions in some strange bracekts?

Reply:
?

Reply:
DMed